libraries/ESP8266WiFiMesh/examples/HelloMesh/HelloMesh.ino

```diff
@@ -1,50 +1,131 @@
+
+  // (void)meshInstance; // This is useful to remove a "unused parameter" compiler warning. Does nothing else.
+  return statusCode;
 }
 
 void setup() {
+  // Prevents the flash memory from being worn out, see: https://github.com/esp8266/Arduino/issues/1054 .
+  // This will however delay node WiFi start-up by about 700 ms. The delay is 900 ms if we otherwise would have stored the WiFi network we want to connect to.
+  WiFi.persistent(false);
+
   Serial.begin(115200);
-  delay(10);
+  delay(50); // Wait for Serial.
+
+  //yield(); // Use this if you don't want to wait for Serial.
 
   Serial.println();
   Serial.println();
+
+  Serial.println("Note that this library can use static IP:s for the nodes to speed up connection times.\n"
+                 "Use the setStaticIP method as shown in this example to enable this.\n"
+                 "Ensure that nodes connecting to the same AP have distinct static IP:s.\n"
+                 "Also, remember to change the default mesh network password!\n\n");
+
   Serial.println("Setting up mesh node...");
 
   /* Initialise the mesh node */
-  mesh_node.begin();
+  meshNode.begin();
+  meshNode.activateAP(); // Each AP requires a separate server port.
+  meshNode.setStaticIP(IPAddress(192, 168, 4, 22)); // Activate static IP mode to speed up connection times.
 }
 
+int32_t timeOfLastScan = -10000;
 void loop() {
-  /* Accept any incoming connections */
-  mesh_node.acceptRequest();
-
-  /* Scan for other nodes and send them a message */
-  char request[60];
-  sprintf(request, "Hello world request #%d from Mesh_Node%d.", request_i++, ESP.getChipId());
-  mesh_node.attemptScan(request);
-  delay(1000);
+  if (millis() - timeOfLastScan > 3000 // Give other nodes some time to connect between data transfers.
+      || (WiFi.status() != WL_CONNECTED && millis() - timeOfLastScan > 2000)) { // Scan for networks with two second intervals when not already connected.
+    String request = "Hello world request #" + String(requestNumber) + " from " + meshNode.getMeshName() + meshNode.getNodeID() + ".";
+    meshNode.attemptTransmission(request, false);
+    timeOfLastScan = millis();
+
+    if (ESP8266WiFiMesh::latestTransmissionOutcomes.empty()) {
+      Serial.println("No mesh AP found.");
+    } else {
+      for (TransmissionResult &transmissionResult : ESP8266WiFiMesh::latestTransmissionOutcomes) {
+        if (transmissionResult.transmissionStatus == TS_TRANSMISSION_FAILED) {
+          Serial.println("Transmission failed to mesh AP " + transmissionResult.SSID);
+        } else if (transmissionResult.transmissionStatus == TS_CONNECTION_FAILED) {
+          Serial.println("Connection failed to mesh AP " + transmissionResult.SSID);
+        } else if (transmissionResult.transmissionStatus == TS_TRANSMISSION_COMPLETE) {
+          // No need to do anything, transmission was successful.
+        } else {
+          Serial.println("Invalid transmission status for " + transmissionResult.SSID + "!");
```

Comment:
this should be an assert, because if the condition ever happens it's due to a programming error or mem corruption.

Reply:
The text of these prints should be moved to flash